packages/input/src/index.ts

```diff
@@ -50,7 +58,11 @@ export default createPrompt<string, InputConfig>((config, done) => {
     if (isEnterKey(key)) {
       const answer = value || defaultValue;
       setStatus('loading');
-
+      if (pattern && !pattern.test(answer)) {
+        setError(patternError || 'Invalid input update');
+        setStatus('idle');
+        return;
```

Comment:
Let's not use an early return and loop the validation within the rest of the logical flow.